### PR TITLE
Intermediate menu solution with hamburger menu

### PR DESCRIPTION
**Pull request by LoredanaChiuta**
- added hamburger menu back
- edited links in the hamburger menu

### CODE REVIEW

File: assets/sass/patterns/molecules/main-menu.scss
Comment (discodavey) [inline assets/sass/patterns/molecules/main-menu.scss:102]: Please roundup or down the half pixel please

File: assets/sass/patterns/molecules/site-header-nav-bar-primary.scss
Comment (discodavey) [inline assets/sass/patterns/molecules/site-header-nav-bar-primary.scss:87]: Can we put the background position/repeat into one line please

```background: url("../../assets/img/patterns/molecules/nav-primary-menu-ic.svg") no-repeat 50px -1px;```

Comment (LoredanaChiuta) [reply assets/sass/patterns/molecules/site-header-nav-bar-primary.scss:87]: I put in a line background position/repetition

**Comment by discodavey** [comment]
@LoredanaChiuta Have you done an internal review of the code as yet?

**Comment by LoredanaChiuta** [comment]
@discodavey The internal review has ended.

File: assets/sass/_mixins.scss
Comment (discodavey) [inline assets/sass/_mixins.scss:478]: background-image and background-repeat can be reduced into one line

**Review by discodavey — COMMENTED** [review]
Two hamburger menu's showing on nav-linked-item.mustache

File: assets/sass/_mixins.scss
Comment (LoredanaChiuta) [reply assets/sass/_mixins.scss:478]: I made the change

**Review by discodavey — COMMENTED** [review]
patterns/02-organisms-global-site-header/02-organisms-global-site-header.html
/patterns/02-organisms-global-site-header-logged-in/02-organisms-global-site-header-logged-in.html

Nav secondary spacing between search, alerts, submit my research is not correct

**Review by discodavey — COMMENTED** [review]
Menu and the hamburger menu does not align correctly. Speak to Chris on how he wants this.

Screenshot 2022-09-20 at 10 39 54

**Comment by discodavey** [comment]
Close and the X seems to be far apart. @chris-huggins Do you want these closer together?

Screenshot 2022-09-20 at 10 41 52

**Review by discodavey — COMMENTED** [review]
Hamburger icon still not looking aligned. @chris-huggins do you agree or do you think I am being too fussy?

Screenshot 2022-09-22 at 09 17 15

**Comment by discodavey** [comment]
@LoredanaChiuta I have looked at your changes and all I can see is the menu issue which is above but I am happy with it otherwise

**Comment by nlisgo** [comment]
We need a review from @chris-huggins or @QueenKraken on this before we approve.

**Comment by chris-huggins** [comment]
@LoredanaChiuta Apologies for not spotting this sooner. Could we please increase the top margin in the hamburger menu above 900px so that the logo and "Close" are aligned with the logo and Home/menu buttons in the main nav?

I think changing the top margin on `.main-menu__container` from 30px to 42px would do it. But please keep it the same as it is now below 900px.
image

**Comment by QueenKraken** [comment]
Related ticket: https://github.com/elifesciences/issues/issues/7697

**Comment by QueenKraken** [comment]
@LoredanaChiuta Thank you - I am happy, once Chris's comments have been addressed.

**Comment by LoredanaChiuta** [comment]
@QueenKraken  The requested change is ready.

**Review by nlisgo — CHANGES_REQUESTED** [review]
I've had a chat with @discodavey and he agreed that are a few things here still to address. I will leave him to complete a full review tomorrow morning.

File: assets/sass/patterns/organisms/site-header.scss
Comment (nlisgo) [inline assets/sass/patterns/organisms/site-header.scss:123]: Surely this comment can go. It seems to correspond to the style that has been removed. `site-header__title:after`

**Review by nlisgo — CHANGES_REQUESTED** [review]

File: assets/sass/patterns/molecules/site-header-nav-bar-primary.scss
Comment (discodavey) [inline assets/sass/patterns/molecules/site-header-nav-bar-primary.scss:120]: Does this need to be used here and not on nav-primary__item--list? Flex can be used once and should filter through the child elements.

**Review by discodavey — COMMENTED** [review]
This does not seem right as there is flex on some of the li's but not all. Flex should be able to be put on the parent (ul) and it should filter down into the children.

Screenshot 2022-09-27 at 10 15 46

**Comment by discodavey** [comment]
I know that @nlisgo wanted to pair on this review with the changes but he is on holiday today and part of tomorrow.

File: source/_patterns/02-organisms/global/site-header.mustache
Comment (LoredanaChiuta) [reply source/_patterns/02-organisms/global/site-header.mustache:9]: I think this is not relevant, because I'm reusing "molecules-site-header-title" in a lot of components and I need that class "site-header__border" only in that file. If I move that class name to another element inside that header title, it will affect all of the component, using "molecules-site-header-title".

Comment (discodavey) [reply source/_patterns/02-organisms/global/site-header.mustache:9]: I will have a look at this later as I do not really want a span that is empty.

**Comment by chris-huggins** [comment]
@discodavey The left and right sections of menu don't appear to be aligned any more. The dividing line is 2px longer than it should be (should be 24px tall), and the space either side of the dividing line isn't equal. I think the space between dividing line and logo should be 4px larger. And the space between dividing line and the menu icon should be 3px less.

**Comment by nlisgo** [comment]
> @discodavey The left and right sections of menu don't appear to be aligned any more. The dividing line is 2px longer than it should be (should be 24px tall), and the space either side of the dividing line isn't equal. I think the space between dividing line and logo should be 4px larger. And the space between dividing line and the menu icon should be 3px less.

@LoredanaChiuta and @discodavey can you confirm that this has been addressed so that @chris-huggins can give his 👍 ?

**Comment by LoredanaChiuta** [comment]
@nlisgo I saw a comment from Dave here https://github.com/elifesciences/issues/issues/7697, so I think that the change is made, but I'm not sure, I think he is able to answer this.

**Comment by chris-huggins** [comment]
> > [elifesciences/issues#7697](https://github.com/elifesciences/issues/issues/7697)
> 
> @chris-huggins can you confirm that this is no longer an issue please?

@nlisgo It does not look like these changes have been made yet.

**Comment by nlisgo** [comment]
> @discodavey The left and right sections of menu don't appear to be aligned any more. The dividing line is 2px longer than it should be (should be 24px tall), and the space either side of the dividing line isn't equal. I think the space between dividing line and logo should be 4px larger. And the space between dividing line and the menu icon should be 3px less.

@LoredanaChiuta this feedback still needs to be addressed.

**Comment by discodavey** [comment]
> > @discodavey The left and right sections of menu don't appear to be aligned any more. The dividing line is 2px longer than it should be (should be 24px tall), and the space either side of the dividing line isn't equal. I think the space between dividing line and logo should be 4px larger. And the space between dividing line and the menu icon should be 3px less.
> 
> @LoredanaChiuta this feedback still needs to be addressed.

When I did the changes I did screen shots and my changes matched up with what had already been done.

**Comment by LoredanaChiuta** [comment]
@nlisgo @chris-huggins @discodavey  I made the last change requested by Chris.

**Comment by nlisgo** [comment]
@chris-huggins can you check again please? If it doesn't match up with what you expect then maybe a quick call could help.

**Comment by chris-huggins** [comment]
@nlisgo Sure I'll check after standup

**Comment by nlisgo** [comment]
@chris-huggins has now indicated he is happy with this. @discodavey has requested that I review this which I am happy to do. Going to discuss with @viktoria-mahmud to see what support she can give.

**Comment by nlisgo** [comment]
@viktoria-mahmud and I had a productive meeting and agreed that @viktoria-mahmud can conduct another review and pair with @LoredanaChiuta before I perform a final review. There are lots of changes to review here and it isn't trivial to assess the impact on downstream applications so a review is quite intensive. Hopefully we can ship this early next week.

File: source/_patterns/00-atoms/components/nav-linked-item.mustache
Comment (nlisgo) [inline source/_patterns/00-atoms/components/nav-linked-item.mustache:3]: I'm not sure that we need the loginControl any more. We can create a followup ticket to clean this up?

File: source/_patterns/01-molecules/navigation/main-menu.yaml
Comment (nlisgo) [inline source/_patterns/01-molecules/navigation/main-menu.yaml:18]: Have submitted a PR to adjust approach on this: https://github.com/elifesciences/pattern-library/pull/1172

File: source/_patterns/01-molecules/navigation/site-header-nav-bar~secondary.json
Comment (nlisgo) [inline source/_patterns/01-molecules/navigation/site-header-nav-bar~secondary.json:9]: We will need to potentially adjust the ViewModel to all for a custom class in this object for each linkedItem

File: source/_patterns/01-molecules/navigation/site-header-title~disable-media-query.json
Comment (nlisgo) [inline source/_patterns/01-molecules/navigation/site-header-title~disable-media-query.json:2]: I believe we can now remove this variant entirely.

**Review by nlisgo — CHANGES_REQUESTED** [review]
I think we are really close. I have read through everything. I have one PR which I think we should get in, a variant of `site-header-title` which we should remove, at least one followup (low priority) ticket to create.

Once we merge this in we need to act quickly on the patterns-php PR otherwise we will be blocking other things getting into pattern-library as these styles will not preserve the current presentation of the navigation. (this is to be expected).

@discodavey can you give the sass some attention please.

File: source/_patterns/01-molecules/navigation/site-header-nav-bar~secondary.json
Comment (LoredanaChiuta) [reply source/_patterns/01-molecules/navigation/site-header-nav-bar~secondary.json:9]: Is there anything to be done about this comment? Can you add more details please?

File: assets/sass/patterns/molecules/main-menu.scss
Comment (discodavey) [inline assets/sass/patterns/molecules/main-menu.scss:31]: Class needs to be renamed to something more relevant. @nlisgo will talk to you about this today

File: source/_patterns/01-molecules/navigation/main-menu.mustache
Comment (discodavey) [inline source/_patterns/01-molecules/navigation/main-menu.mustache:13]: main-menu__list_change also needs to be renamed as well. @nlisgo will talk to you about this also.

**Review by discodavey — COMMENTED** [review]
The eLife logo needs to be outside of the nav tag as it's not part of the navigation. Same goes for the Close button

Screenshot 2022-10-05 at 09 04 47

File: source/_patterns/01-molecules/navigation/main-menu.mustache
Comment (nlisgo) [reply source/_patterns/01-molecules/navigation/main-menu.mustache:13]: The class names have been changed and we have swapped the `isHiddenMainMenuItemForWideScreen` flag with a `classes` property.

File: assets/sass/patterns/molecules/main-menu.scss
Comment (nlisgo) [reply assets/sass/patterns/molecules/main-menu.scss]: The class names have been changed and we have swapped the `hasBorderBottom ` flag with a `classes` property.

**Comment by LoredanaChiuta** [comment]
> The eLife logo needs to be outside of the nav tag as it's not part of the navigation. Same goes for the Close button
> 
> Screenshot 2022-10-05 at 09 04 47

the close button is generated with js and I will have to change it there, is it ok?

**Review by discodavey — COMMENTED** [review]
As long as the logo and the close button are moved outside the <nav> tag this is fine

File: source/_patterns/01-molecules/navigation/site-header-nav-bar~secondary.json
Comment (nlisgo) [inline source/_patterns/01-molecules/navigation/site-header-nav-bar~secondary.json:15]: Another followup ticket to address how we can add a custom class to a nav item. Current solution is to check for the word alert in the text.

File: source/_patterns/00-atoms/components/nav-linked-item.mustache
Comment (nlisgo) [reply source/_patterns/00-atoms/components/nav-linked-item.mustache:3]: No action needed. I will create a ticket.

**Review by nlisgo — CHANGES_REQUESTED** [review]
Thanks for the change to the navigation. We now just need to address the failing test. If you need support with this then please reach out.

**Comment by LoredanaChiuta** [comment]
> Thanks for the change to the navigation. We now just need to address the failing test. If you need support with this then please reach out.

@nlisgo I manually changed the file to pass the tests.

**Review by nlisgo — APPROVED** [review]

**Pull request merged**
